lib/mysql.js

```diff
@@ -412,7 +412,7 @@ MySQL.prototype.fromColumnValue = function(prop, val) {
         // MySQL allows, unless NO_ZERO_DATE is set, dummy date/time entries
         // new Date() will return Invalid Date for those, so we need to handle
         // those separate.
-        if (!val || val == '0000-00-00 00:00:00' || val == '0000-00-00') {
+        if (!val || /^0{4}(-00){2}( (00:){2}0{2}(\.0{1,6}){0,1}){0,1}$/.test(val)) {
```

Comment:
This regex checks if val is one of the following values:

0000-00-00
0000-00-00 00:00:00
0000-00-00 00:00:00.0
0000-00-00 00:00:00.00
0000-00-00 00:00:00.000
0000-00-00 00:00:00.0000
0000-00-00 00:00:00.00000
0000-00-00 00:00:00.000000

You can see it working over here: https://regex101.com/r/IkM0wc/2